lib/logger.js

```diff
@@ -148,13 +151,16 @@ class Logger {
    */
 
   async truncate() {
+    if (this.truncating)
+      return;
+
     if (!this.filename)
       return;
 
     if (fs.unsupported)
       return;
 
-    assert(!this.stream);
+    this.truncating = true;
 
```

Comment:
Reading through the control flow:

`truncating` is set to true on line 163
It is then set to false in the `finally` clause on line 173
Will `truncating` ever be `true` on line 180? Reading the control flow, it doesn't look like it. Is setting it to `false` necessary?

Reply:
huh good catch, I'll try taking it out

Reply:
That should probably be inside the catch instead.

Reply:
updated.